src/adapter/src/session.rs

```diff
@@ -978,8 +993,15 @@ impl<T: TimestampManipulation> TransactionStatus<T> {
                         }
                         *ops = add_ops;
                     }
-                    TransactionOps::Peeks(determination) => match add_ops {
-                        TransactionOps::Peeks(add_timestamp_determination) => {
+                    TransactionOps::Peeks {
+                        determination,
+                        cluster_id,
+                    } => match add_ops {
+                        TransactionOps::Peeks {
+                            determination: add_timestamp_determination,
+                            cluster_id: add_cluster_id,
+                        } => {
+                            assert_eq!(*cluster_id, add_cluster_id);
```

Comment:
instead of an `assert_eq!` can this be an error, since this function is documented as "An error is produced if they cannot be merged"

Reply:
There are two asserts below that do the same thing. I don't recall from reading over this WHY I wrote assert instead of returning an error during merge.

Edit: oh yeah! The caller is responsible for enforcing this because it should have already checked if the txn is an in-progress Peek txn and use its timestamp (and now cluster id) and do verification there. We should probably document exactly that, because the current fn docs are misleading.

Reply:
Sorry, I missed this. I'll submit a follow up PR.

Reply:
I think it should stay as an assert!